### PR TITLE
Adds CocinaObjectPersister.

### LINKED ISSUE #3143: Create cocina object persister service
The role of the service is to support finding and saving Cocina objects. The service (called `CocinaObjectPersister`) will have the following methods and initial implementations:
* `save`: Save will retrieve the Fedora item, perform an update with Cocina::ObjectUpdater, and save the item. For reference, see ObjectsController.update.
* `find`: Find will retrieve the Fedora item and map to a Cocina object with Cocina::Mapper. For reference, see ObjectsController.show.

As DSA services are updated to operate on Cocina objects instead of Fedora objects, CocinaObjectPersister will be used to abstract finding and saving.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #3143

## Why was this change made?
Provide an abstraction around finding / saving Cocina objects.

## How was this change tested?
Unit

### CODE REVIEW

File: app/controllers/objects_controller.rb [hunk 1]
```diff
@@ -40,14 +57,11 @@ def create
   end
 
   def update
```
Comment (justinlittman) [inline app/controllers/objects_controller.rb:59]: update switched to use CocinaObjectPersister.

File: app/controllers/objects_controller.rb [hunk 2]
```diff
@@ -56,21 +70,8 @@ def update
   end
 
   def show
```
Comment (justinlittman) [inline app/controllers/objects_controller.rb:72]: show (via load_cocina before action) switched to use CocinaObjectPersister.

File: app/controllers/objects_controller.rb [hunk 3]
```diff
@@ -2,13 +2,15 @@
...
-  before_action :load_item, except: [:create]
+  before_action :load_cocina_object, only: %i[show update_doi_metadata]
+  before_action :load_item, only: %i[show accession update_marc_record destroy notify_goobi]
 
+  # No longer be necessary when remove Fedora.
   rescue_from(Cocina::ObjectUpdater::NotImplemented) do |e|
     json_api_error(status: :unprocessable_entity, message: e.message)
   end
 
-  rescue_from(ActiveFedora::ObjectNotFoundError) do |e|
```
Comment (jcoyne) [inline app/controllers/objects_controller.rb:11]: I don't think you want to remove this clause until there are no more usages of `Dor.find` (e.g. `load_item`)

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/controllers/objects_controller.rb (hunk 3 above)
Comment (justinlittman) [reply app/controllers/objects_controller.rb:11]: Good call. Fixed.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -61,4 +61,8 @@ def http_auth_header
   def load_item
     @item = Dor.find(params[:object_id] || params[:id])
   end
+
+  def load_cocina_object
+    @cocina_object = CocinaObjectPersister.find(params[:object_id] || params[:id])
```
Comment (jcoyne) [inline app/controllers/application_controller.rb:66]: I'm wondering about the name "persister".  this is also doing loading. Can we change the name of the class to CocinaObjectRepository?

**Review by jcoyne — APPROVED** [review]
Approved pending consideration of class name change.

File: app/controllers/application_controller.rb (hunk 4 above)
Comment (justinlittman) [reply app/controllers/application_controller.rb:66]: Renaming to CocinaObjectStore (prefer over Repository).

Comment (jcoyne) [reply app/controllers/application_controller.rb]: That works for me.

**Pull request merged**